### PR TITLE
refactor: ml model load read from class type hints

### PR DESCRIPTION
**Pull request by GarrettWu**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: bigframes/ml/cluster.py [hunk 1]
```diff
@@ -34,6 +34,7 @@
     "distance_type": "distanceType",
     "max_iter": "maxIterations",
     "tol": "minRelativeProgress",
+    "warm_start": "warmStart",
```
Comment (ashleyxuu) [inline bigframes/ml/cluster.py:37]: We should add the unit test for all these newly-added params. Same for all other BQML models, but feel free to address in a follow-up PR.

File: bigframes/ml/decomposition.py [hunk 2]
```diff
@@ -27,6 +27,8 @@
 from bigframes.ml import base, core, globals, utils
 import bigframes.pandas as bpd
 
+_BQML_PARAMS_MAPPING = {"svd_solver": "pcaSolver"}
```
Comment (ashleyxuu) [inline bigframes/ml/decomposition.py:30]: Same here.

Comment (GarrettWu) [reply bigframes/ml/decomposition.py:30]: This one isn't a new adding, but moved piece.

File: bigframes/ml/cluster.py (hunk 1 above)
Comment (GarrettWu) [reply bigframes/ml/cluster.py:37]: yes, will do a follow up.

**Review by ashleyxuu — APPROVED** [review]

**Pull request merged**